indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx

```diff
@@ -106,43 +99,53 @@ export default function ReviewForm() {
     );
 
   const judgmentForm = (
-    <div className="flexrow f-a-center" styleName="judgment-form">
-      <CommentForm onSubmit={createComment} onToggleExpand={setCommentFormVisible} />
-      {canPerformSubmitterActions && canReview && !editor && (
-        <>
-          <span className="comment-or-review">
-            <Translate>or</Translate>
-          </span>
-          <EditableSubmissionButton
-            eventId={eventId}
-            contributionId={contribution.id}
-            contributionCode={contribution.code}
-            fileTypes={{[editableType]: fileTypes}}
-            uploadableFiles={lastRevisionWithFiles.files}
-            text={Translate.string('Submit files')}
-            onSubmit={handleSubmission}
-          />
-        </>
-      )}
-      {!commentFormVisible && canJudge && (
-        <div className="review-trigger flexrow">
-          <span className="comment-or-review">
-            <Translate>or</Translate>
-          </span>
-          <Dropdown
-            className="judgment-btn"
-            text={Translate.string('Judge')}
-            direction="left"
-            button
-            floating
-          >
-            <Dropdown.Menu>
-              <JudgmentDropdownItems options={judgmentOptions} setJudgmentType={setJudgmentType} />
-            </Dropdown.Menu>
-          </Dropdown>
+    <Popup
+      trigger={
+        <div className="flexrow f-a-center" styleName="judgment-form">
+          <CommentForm onSubmit={createComment} onToggleExpand={setCommentFormVisible} />
```

Comment:
I can't find a way to pass a `disabled` prop to the `CommentForm` without resetting its state...

Reply:
Isn't this what you need here? (didn't bother with proptypes to test this ;))

```patch
diff --git a/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx b/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx
index 2378acd0ba..0da6e66771 100644
--- a/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx
+++ b/indico/modules/events/editing/client/js/editing/timeline/CommentForm.jsx
@@ -18,7 +18,7 @@ import {getDetails} from './selectors';

 import './CommentForm.module.scss';

-export default function CommentForm({onSubmit, onToggleExpand, initialValues, expanded}) {
+export default function CommentForm({onSubmit, onToggleExpand, initialValues, expanded, disabled}) {
   const {canCreateInternalComments} = useSelector(getDetails);
   const [commentFormVisible, setCommentFormVisible] = useState(expanded);

@@ -67,7 +67,7 @@ export default function CommentForm({onSubmit, onToggleExpand, initialValues, ex
                   />
                 )}
                 <Form.Group styleName="submit-buttons" inline>
-                  <FinalSubmitButton label={Translate.string('Comment')} />
+                  <FinalSubmitButton label={Translate.string('Comment')} disabled={disabled} />
                   <Button
                     disabled={fprops.submitting}
                     content={Translate.string('Cancel')}
diff --git a/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx b/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx
index 0817eddea6..2be19abfe6 100644
--- a/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx
+++ b/indico/modules/events/editing/client/js/editing/timeline/ReviewForm.jsx
@@ -102,7 +102,11 @@ export default function ReviewForm() {
     <Popup
       trigger={
         <div className="flexrow f-a-center" styleName="judgment-form">
-          <CommentForm onSubmit={createComment} onToggleExpand={setCommentFormVisible} />
+          <CommentForm
+            onSubmit={createComment}
+            onToggleExpand={setCommentFormVisible}
+            disabled={isOutdated}
+          />
           {canPerformSubmitterActions && canReview && !editor && (
             <>
               <span className="comment-or-review">
```

Reply:
thats what I did, at least for me this clears the form when the timeline becomes outdated. not sure why